iscsi/get.go

```diff
@@ -0,0 +1,283 @@
+// Copyright 2017 The Prometheus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iscsi
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+// GetStats is the main iscsi status information func
+// building the path and prepare info for enable iscsi
+func GetStats(iqnPath string) (*Stats, error) {
+	var istats Stats
+
+	istats.Name = filepath.Base(iqnPath)
+	istats.RootPath = filepath.Dir(iqnPath)
+
+	matches, err := filepath.Glob(filepath.Join(iqnPath, "tpgt*"))
+	if err != nil {
+		return nil, fmt.Errorf("iscsi: GetStats: get TPGT path error %v", err)
+	}
+	istats.Tpgt = make([]TPGT, len(matches))
+
+	for i, tpgtPath := range matches {
+		istats.Tpgt[i].Name = filepath.Base(tpgtPath)
+		istats.Tpgt[i].TpgtPath = tpgtPath
+		istats.Tpgt[i].IsEnable, _ = isPathEnable(tpgtPath)
+		if istats.Tpgt[i].IsEnable {
+			matchesLunsPath, _ := filepath.Glob(filepath.Join(tpgtPath, "lun/lun*"))
+
+			for _, lunPath := range matchesLunsPath {
+				lun, err := getLunLinkTarget(lunPath)
+				if err == nil {
+					istats.Tpgt[i].Luns = append(istats.Tpgt[i].Luns, lun)
+				}
+			}
+		}
+	}
+	return &istats, nil
+}
+
+// isPathEnable is a utility function
+// check if the file "enable" contain enable message
+func isPathEnable(path string) (bool, error) {
+	var isEnable bool
+	isEnable = false
+	enableReadout, err := ioutil.ReadFile(filepath.Join(path, "enable"))
+	if err != nil {
+		return false, fmt.Errorf("iscsi: isPathEnable error %v", err)
+	}
+	enableReadoutNum, err := strconv.Atoi(strings.TrimSpace(string(enableReadout)))
```

Comment:
Instead of converting to an integer and then interpreting the integer, I think you can use `strings.ParseBool` which will interpret `0` as false and `1` as true.

Reply:
Thanks, yeah replace all the un-needed variable and test with strconv.ParseBool